gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/AuditEnvironmentUpgrader.java

```diff
@@ -0,0 +1,134 @@
+
+    private final EnvironmentRepository environmentRepository;
+
+    @Autowired
+    public AuditEnvironmentUpgrader(
+        AuditRepository auditRepository,
+        ApiRepository apiRepository,
+        ApplicationRepository applicationRepository,
+        EnvironmentRepository environmentRepository
+    ) {
+        super(InstallationService.AUDIT_ENVIRONMENT_UPGRADER);
+        this.auditRepository = auditRepository;
+        this.apiRepository = apiRepository;
+        this.applicationRepository = applicationRepository;
+        this.environmentRepository = environmentRepository;
+    }
+
+    @Override
+    public int getOrder() {
+        return 500;
+    }
+
+    @Override
+    protected void processOneShotUpgrade(ExecutionContext executionContext) throws TechnicalException {
+        auditRepository
+            .findAll()
+            .forEach(
+                audit -> {
+                    try {
+                        switch (audit.getReferenceType()) {
+                            case API:
+                                handleApiAudit(audit);
+                                break;
+                            case APPLICATION:
+                                handleApplicationAudit(audit);
+                                break;
+                            case ENVIRONMENT:
+                                handleEnvironmentAudit(audit);
+                                break;
+                            case ORGANIZATION:
+                                handleOrganizationAudit(audit);
+                                break;
+                        }
+                    } catch (AbstractNotFoundException | TechnicalException e) {
+                        LOGGER.error("Failed to update audit {}", audit.getId(), e);
+                    }
+                }
+            );
+    }
+
+    private void handleApiAudit(Audit audit) throws TechnicalException {
+        Api api = apiRepository.findById(audit.getReferenceId()).orElseThrow(() -> new ApiNotFoundException(audit.getReferenceId()));
+        updateAuditEnvironment(api.getEnvironmentId(), audit);
+    }
+
+    private void handleApplicationAudit(Audit audit) throws TechnicalException {
+        Application application = applicationRepository
+            .findById(audit.getReferenceId())
+            .orElseThrow(() -> new ApplicationNotFoundException(audit.getReferenceId()));
+        updateAuditEnvironment(application.getEnvironmentId(), audit);
+    }
+
+    private void handleEnvironmentAudit(Audit audit) throws TechnicalException {
+        updateAuditEnvironment(audit.getReferenceId(), audit);
+    }
+
+    private void handleOrganizationAudit(Audit audit) throws TechnicalException {
+        audit.setOrganizationId(audit.getReferenceId());
+        auditRepository.update(audit);
+    }
+
+    private void updateAuditEnvironment(String environmentId, Audit audit) throws TechnicalException {
+        Environment environment = environmentRepository
```

Comment:
We should avoid reading environment multiple times, using a map to store them.
For example : https://github.com/gravitee-io/gravitee-api-management/blob/master/gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/SearchIndexUpgrader.java#L145